src/mono/mono/mini/tramp-amd64.c

```diff
@@ -1179,7 +1179,7 @@ mono_arch_get_interp_to_native_trampoline (MonoTrampInfo **info)
 
 	/* save all return floating registers in the CallContext */
 	for (i = 0; i < FLOAT_RETURN_REGS; i++)
-		amd64_sse_movsd_membase_reg (code, AMD64_R11, MONO_STRUCT_OFFSET (CallContext, fregs) + i * sizeof (double), i);
+		amd64_sse_movsd_membase_reg (code, AMD64_R11, MONO_STRUCT_OFFSET (CallContext, fregs) + float_return_regs [i] * sizeof (double), float_return_regs [i]);
```

Comment:
Why is this change necessary only for amd64?

Reply:
On arm64, the argument and return registers are the same (x0-x7). Whereas on amd64, different registers are used for arguments and returns.

For arm64, this gets handled at https://github.com/dotnet/runtime/blob/4185f94ec85e619b7dfa82357ac853846dfda3af/src/mono/mono/mini/tramp-arm64.c#L773-L779